sql/logic_test.go

```diff
@@ -204,27 +204,25 @@ func (t *logicTest) setUser(tempDir, user string) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	dir, err := os.Getwd()
-	if err != nil {
-		t.Fatal(err)
-	}
-	certDir := filepath.Join(filepath.Dir(dir), "resource", security.EmbeddedCertsDir)
 
-	certPath := security.ClientCertPath(certDir, user)
-	keyPath := security.ClientKeyPath(certDir, user)
+	caPath := filepath.Join(security.EmbeddedCertsDir, "ca.crt")
+	certPath := security.ClientCertPath(security.EmbeddedCertsDir, user)
+	keyPath := security.ClientKeyPath(security.EmbeddedCertsDir, user)
 
-	// `github.com/lib/pq` requires that private key file permissions are
-	// "u=rw (0600) or less".
-	tmpKeyPath := tempRestrictedCopy(t.T, keyPath, tempDir)
+	// Copy these assets to disk from embedded strings, so this test can
+	// run from a standalone binary.
+	tempCAPath, _ := tempRestrictedCopy(t.T, tempDir, caPath, "TestLogic_ca")
```

Comment:
Rather than passing in the prefix for the temp file, you could derive it from the source path using `filepath.Base()`.